app/src/main/java/com/android254/droidconke19/datastates/RunCatching.kt

```diff
@@ -0,0 +1,7 @@
+package com.android254.droidconke19.datastates
+
+suspend fun <T, R : Any> T.runCatching(block: suspend T.() -> Result<R>) = try {
+    block()
+} catch (e: Exception) {
+    Result.Error(e.message)
+}
```

Comment:
I think you should change the implementation to the following:
```
suspend fun <R : Any> runCatching(block: suspend () -> Result<R>) =  = withContext(Dispatchers.IO) { 
    return@withContext try {
        Result.Success(block())
    } catch (e: Exception) {
        Result.Error(e.message)
    }
}
```
I don't think the receiver is necessary. Also `Result.Success` should be called from this function. Also, @wangerekaharun had mentioned moving the context specification to this function as well.